### PR TITLE
fix horrifying predict output transpose issue

### LINKED ISSUE #143: problem with XGBoost.predict on booster created with multi:softprob
Am modeling the Framingham Heart Study dataset from Kaggle.  Dataset has been prepared for XGBoost.  Model trained with objective  set to multi:softprob.  When I execute XGBoost.predict against the training data, the probality matrix returned has extremely poor accuracy.  I trained the identical processed dataset using xgboost in R.  The training evaluation mlogloss for both approaches are near identical.  In R, when I 'predict' the probability matrix against the training data I get a completely different set of matrix values.  The accuracy of the predictions on the R output is very high.  Is it possible there is an error in XGBoost.predict for the multi:softprob objective?  It does not appear to be an issue with the reshaping of the data.  Does XGBoost.predict differentiate between output from multi:softmax and multi:softprob - just trying to imagine where the issue could be.

I am happy to send my processed dataset as well as code I've used in Julia and R, just let me know where/how to send.

Thank you.

### PR DESCRIPTION
**Pull request by ExpandingMan**
This fixes #143 in which the output of `predict`, if a matrix, would get scrambled because it failed to transpose the dimensions returned by the row-major libxboost.  Unfortunately this comes at the cost of the type stability of `predict` since now the output may be a `transpose` array (though we assume this is preferable to actually copying it).

I have added a test for classifications that return matrix output from `predict` using test data that I generated with MLJ.  This data has been added to `assets/data`.

Since this is a critical issue I'm probably not going to wait very long to merge this.

### CODE REVIEW

**Comment by bobaronoff** [comment]
Thank you for addressing this so promptly.  I am new to this so pardon if this is a silly comment.  Noticed in your code you added a simple transpose if the return length(dim)>1.  I don't think a simple transpose will suffice.  I had to reshape the matrix before transposing.  For example the 'old' return matrix was (920,5) - a column for each class.  That is correct dimension.  A transpose will give a (5,920) matrix - not helpful.  I first needed to reshape the (920,5) to (5,920) and then apply the transpose.  I think this is correct but apologize if I am wrong.

regards,
Bob

**Comment by ExpandingMan** [comment]
There's a `transpose` but it also reverses the dimensions that are output to be used in the matrix constructor.  That's why you weren't able to fix it with a transpose alone.
